common/azure-common/src/test/java/com/azure/common/implementation/RestProxyWithMockTests.java

```diff
@@ -413,4 +422,183 @@ public HttpClient port(int port) {
             throw new IllegalStateException("MockHttpClient.port not implemented.");
         }
     }
+
+
+    static class KeyValue {
+        @JsonProperty("key")
+        private int key;
+
+        @JsonProperty("value")
+        private String value;
+
+        KeyValue() { }
+        KeyValue(int key, String value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        int key() {
+            return this.key;
+        }
+
+        String value() {
+            return this.value;
+        }
+    }
+
+    static class KeyValuePage implements Page<KeyValue> {
+        @JsonProperty()
+        private List<KeyValue> items;
+
+        @JsonProperty("nextLink")
+        private String nextLink;
+
+        KeyValuePage() {
+        }
+
+        KeyValuePage(List<KeyValue> items, String nextLink) {
+            this.items = items;
+            this.nextLink = nextLink;
+        }
+
+        @Override
+        public List<KeyValue> items() {
+            return items;
+        }
+
+        @Override
+        public String nextLink() {
+            return nextLink;
+        }
+    }
+
+    static class ConformingPage<T> implements Page<T> {
+        private List<T> items;
+        private String nextLink;
+
+        ConformingPage(List<T> items, String nextLink) {
+            this.items = items;
+            this.nextLink = nextLink;
+        }
+
+        @Override
+        public List<T> items() {
+            return items;
+        }
+
+        @Override
+        public String nextLink() {
+            return nextLink;
+        }
+    }
+
+    static class NonComformingPage<T> {
```

Comment:
so we call it `NonComformingPage` because it does not implements `Page` interface?

Reply:
Yeah. I'll add some documentation to that class to explain why. :)